### PR TITLE
chore: Add semantic-release

### PR DESCRIPTION
**Pull request by rylanc**
Enable easy gem releases.

I couldn't think of a good way to test this. I've tested it locally, without actually pushing the gem and things work OK up to that point.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
+      before_install:
+        - rvm use 2.3 --install --binary --fuzzy
+        - ruby --version
+        - gem update --system
+        - gem install bundler
+        - bundle install --jobs=3 --retry=3 --deployment
+      script:
+        - yarn test
+        - yarn lint
+    # TODO: Only add this stage on master?
```
Comment (bagelbits) [inline .travis.yml]: It probably makes sense to only add this stage on master. I think you can add this:
```
branches:
  only:
  - master
```

Comment (rylanc) [reply .travis.yml]: Done!

**Comment by zp-alfred** [comment]
Accepted by bagelbits!

**Pull request merged**